### DISCUSSION TITLE
404 not working on sub routes

### DISCUSSION THREAD

**Discussion by peterpeterparker** [facebook/docusaurus#9664] [Q&A]
Hi, I have swizzled the NotFound component to customize my 404 but, while it works fine on root URL, it isn't used for subroutes.

For example, `http://localhost:3000/unknown` works as expected however `http://localhost:3000/docs/unknown` returns the generic 404.

Same with prod build, e.g. https://juno.build/unknown vs https://juno.build/docs/unknown. With prod build deployed I briefly see my 404 but, it seems that the JS kicks it out. 

Anything I can do to fix this problem?

---

Repo: https://github.com/junobuild/docs

---

Good:

<img width="1535" alt="Capture d’écran 2023-12-22 à 18 23 56" src="https://github.com/facebook/docusaurus/assets/16886711/d716b5c2-4cd9-47a5-8418-05e13331232b">

Not good:

<img width="1535" alt="Capture d’écran 2023-12-22 à 18 23 58" src="https://github.com/facebook/docusaurus/assets/16886711/6f33eacb-0598-4d53-9bc8-3f3a712544cc">

**Comment by slorber** [comment]
Hmmmm yes this looks suspicious

The client-side router seems to route the url to another page for some reason, not sure why.

Can you create a minimal repro and open a bug please?

**Reply by peterpeterparker** [reply]
Sure of course, there you go 👉  https://github.com/facebook/docusaurus/issues/9665

Thanks for the feedback!
